lib/spack/spack/test/spec_yaml.py

```diff
@@ -421,12 +421,13 @@ def test_load_json_specfiles(specfile, expected_hash, reader_cls):
     assert len(openmpi_edges) == 1
 
     # Check that virtuals have been reconstructed
-    assert "mpi" in openmpi_edges[0].virtuals
+    if reader_cls.SPEC_VERSION >= spack.spec.SpecfileV4.SPEC_VERSION:
```

Comment:
We should also change the comment above, cause this change is such that we skip testing virtual reconstruction on edges

Reply:
Great catch. Thank you.

Reply:
Updated with reference to conformance.